### DISCUSSION TITLE
Bulk update child records

### DISCUSSION THREAD

**Discussion by jeepers3327** [prisma/prisma#7978] [Q&A] [answered]
Hi,

I'm trying this ORM on my new project and I would like to ask if there is/are ways to bulk update child records.

### Sample Models
```prisma
model Gist {
  id          String  @id @default(uuid())
  description String
  private     Boolean @default(false)
  author      User?   @relation(fields: [authorId], references: [id])
  authorId    String?

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  gistFiles GistFile[]
  comments  Comment[]
  @@map("gists")
}

model GistFile {
  id String @id @default(uuid())

  gist     Gist?   @relation(fields: [gistId], references: [id])
  gistId   String?
  filename String
  snippet  String
  language String
  @@map("gist_files")
}
```

For example I have these update json payload:
```json
{
    "gist": {
        "description": "updated gist desc",
        "private": true,
        "gistFiles": [
            {
                "id": "5605089e-edc9-45f8-b495-20cb523d83e7",
                "filename": "t.txt",
                "language": "text",
                "snippet": "This is a text file"
            },
             {
                 "id": "d522a7f5-7c90-4f55-a8f7-a2393a67ea1a",
                "filename": "t.ts",
                "language": "typescript",
                "snippet": "const a : number = 1;"
            }
        ]
    }
}
```

How do I bulk update the `gistFiles` which is a child of `gist` entity given its id?
I looked at `updateMany` but it is not I'm looking for as the function only updates a single value for related records. I also looked at `set` but it only works for existing record.

My possible workaround is running a `update` function inside a map.
```ts
gistData.gistFiles.map(async gist => {
   await this.client.gistFile.update({
       data: {...gist},
       where: {
         id: gist.id
      }
   })
});
```

Is there a way using of achieving the same as shown above without resorting to loops/maps? I'd imagined something like this.
```ts
        await this.client.gistFile.updateMany({
            data: [
                {
                    id: "5605089e-edc9-45f8-b495-20cb523d83e7",
                    filename: "t.txt",
                    language: "text",
                    snippet: "This is a textfile"
                },
                 {
                    id: "d522a7f5-7c90-4f55-a8f7-a2393a67ea1a",
                    filename: "t.ts",
                    language: "typescript",
                    snippet: "const a : number = 1;"
                }
            ]
        })
```

Again, thank you for these awesome tool.

**Comment by ryands17** [comment, marked as answer]
@jeepers3327 👋 

Unfortunately this is the only way you can do this at the moment. There's a request for the same here so it would be great if you could add a 👍 to this so that we can set the priority.

https://github.com/prisma/prisma/issues/6862

**Reply by jeepers3327** [reply]
Thanks

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
